src/Consolonia.Core/Infrastructure/InputLessDefaultNetConsole.cs

```diff
@@ -47,21 +35,63 @@ protected InputLessDefaultNetConsole()
         public bool CaretVisible
         {
             get => _caretVisible;
+#pragma warning disable CA1303 // Do not pass literals as localized parameters
             set
             {
                 if (_caretVisible == value) return;
-                Console.CursorVisible = value;
+                WriteText(value ? Esc.ShowCursor : Esc.HideCursor);
                 _caretVisible = value;
             }
+#pragma warning restore CA1303 // Do not pass literals as localized parameters
         }
 
         public PixelBufferSize Size { get; private set; }
 
-        public bool SupportsComplexEmoji { get; }
+        public bool SupportsComplexEmoji
+        {
+            get
+            {
+                if (_supportEmoji == null)
+                {
+#pragma warning disable CA1031 // Do not catch general exception types
+                    try
+                    {
+                        // Detect complex emoji support by writing a complex emoji and checking cursor position.
+                        // If the cursor moves 2 positions, it indicates proper rendering of composite surrogate pairs.
+                        (int left, int top) = Console.GetCursorPosition();
+                        WriteText(
+                            $"{Esc.Foreground(Colors.Transparent)}{Esc.Background(Colors.Transparent)}{TestEmoji}");
+
+                        // TODO, escape sequence
+                        (int left2, _) = Console.GetCursorPosition();
+                        _supportEmoji = left2 - left == 2;
+                        Console.SetCursorPosition(left, top);
+                    }
+                    catch (Exception)
```

Comment:
Same question. If for instance OutOfMemoryException happened, why are we assuming emoji are supported?

Reply:
I removed this, I no longer need to do this.